src/client/auth/oidc.rs

```diff
@@ -144,7 +157,7 @@ impl Callback {
 
     /// Create azure callback.
     #[cfg(feature = "azure-oidc")]
-    fn azure_callback(client_id: Option<&str>, resource: &str) -> CallbackInner {
+    fn azure_callback(client_id: Option<&str>, resource: &str) -> Function {
```

Comment:
This signature change wasn't needed for the functionality here, it just gets rid of a level of indentation and means each of the `*_callback` functions doesn't need to construct its own fresh cache that can be done just as easily at the call site.